src/main.rs

```diff
@@ -1,98 +1,173 @@
-use std::convert::TryInto;
+use std::env;
 
-const BLOCK_SIZE: usize = 32; // Size of each block in bits
-const HASH_SIZE: usize = 32; // Size of the hash code in bits
+const BLOCK_SIZE: usize = 8; // Size of each block in bytes
+const HASH_SIZE: usize = 8; // Size of the hash code in bytes
```

Comment:
If these are ever not the same value a lot of things will break. Look into either combining these constants into one or fixing the places that will break.

Reply:
Maybe this is what `finalize()` is supposed to do.